File: src/main/java/com/refinedmods/refinedstorage/render/model/DelegateBakedModel.java
package com.refinedmods.refinedstorage.render.model;

import com.mojang.blaze3d.vertex.PoseStack;
import net.minecraft.client.renderer.block.model.BakedQuad;
import net.minecraft.client.renderer.block.model.ItemOverrides;
import net.minecraft.client.renderer.block.model.ItemTransforms;
import net.minecraft.client.renderer.texture.TextureAtlasSprite;
import net.minecraft.client.resources.model.BakedModel;
import net.minecraft.core.Direction;
import net.minecraft.world.level.block.state.BlockState;

import javax.annotation.Nullable;
import java.util.List;
import java.util.Random;

public class DelegateBakedModel implements BakedModel {
    protected final BakedModel base;

    public DelegateBakedModel(BakedModel base) {
        this.base = base;
    }

    @Override
    @SuppressWarnings("deprecation")
    public List<BakedQuad> getQuads(@Nullable BlockState state, @Nullable Direction side, Random rand) {
        return base.getQuads(state, side, rand);
    }

    @Override
    public boolean useAmbientOcclusion() {
        return base.useAmbientOcclusion();
    }

    @Override
    public boolean isGui3d() {
        return base.isGui3d();
    }

    @Override
    public boolean usesBlockLight() {
        return base.usesBlockLight();
    }

    @Override
    public boolean isCustomRenderer() {
        return base.isCustomRenderer();
    }

    @Override
    @SuppressWarnings("deprecation")
    public TextureAtlasSprite getParticleIcon() {
        return base.getParticleIcon();
    }

    @Override
    public ItemOverrides getOverrides() {
        return base.getOverrides();
    }

    @Override
    @SuppressWarnings("deprecation")
    public ItemTransforms getTransforms() {
        return base.getTransforms();
    }

    @Override
    @SuppressWarnings("deprecation")
    public BakedModel handlePerspective(ItemTransforms.TransformType cameraTransformType, PoseStack matrixStack) {
        return base.handlePerspective(cameraTransformType, matrixStack);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/render/model/DelegateBakedModel.kt
package com.refinedmods.refinedstorage.render.model

import com.mojang.blaze3d.matrix.MatrixStack
import net.minecraft.block.BlockState
import net.minecraft.client.renderer.model.BakedQuad
import net.minecraft.client.renderer.model.IBakedModel
import net.minecraft.client.renderer.model.ItemCameraTransforms
import net.minecraft.client.renderer.model.ItemOverrideList
import net.minecraft.client.renderer.texture.TextureAtlasSprite
import net.minecraft.util.Direction
import java.util.*

open class DelegateBakedModel(base: IBakedModel) : IBakedModel {
    protected val base: IBakedModel
    open fun getQuads(@Nullable state: BlockState?, @Nullable side: Direction, rand: Random): List<BakedQuad?>? {
        return base.getQuads(state, side, rand)
    }

    val isAmbientOcclusion: Boolean
        get() = base.isAmbientOcclusion()
    val isGui3d: Boolean
        get() = base.isGui3d()

    fun func_230044_c_(): Boolean {
        return base.func_230044_c_()
    }

    val isBuiltInRenderer: Boolean
        get() = base.isBuiltInRenderer()
    val particleTexture: TextureAtlasSprite
        get() = base.getParticleTexture()
    open val overrides: ItemOverrideList
        get() = base.getOverrides()
    val itemCameraTransforms: ItemCameraTransforms
        get() = base.getItemCameraTransforms()

    fun handlePerspective(cameraTransformType: ItemCameraTransforms.TransformType?, matrixStack: MatrixStack?): IBakedModel {
        return base.handlePerspective(cameraTransformType, matrixStack)
    }

    init {
        this.base = base
    }
}